server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -151,10 +156,15 @@ public boolean parsesArrayValue() {
     /**
      * Parse the field value using the provided {@link ParseContext}.
      */
-    public void parse(ParseContext context) throws IOException {
+    public final void parse(ParseContext context) throws IOException {
         try {
-            parseCreateField(context);
-        } catch (Exception e) {
+            valueParser.parseAndIndex(context.parser(), v -> {
```

Comment:
I think multi-fields should parse their own values instead of consuming the already-parsed ones. For example, if a `text` field is a sub-field of a `keyword`, then it should see the full original value instead of the value after `normalizer` and `ignore_above` have been applied.

I also think the behavior is simpler if multi-fields parse source values in exactly the same way as when they're root fields, instead having alternate logic. (I realize we already have a discrepancy around 'external values', but I would like to remove support for that!)